openmessaging-api/src/main/java/io/openmessaging/interceptor/ProducerInterceptor.java

```diff
@@ -1,50 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
- */
-
-package io.openmessaging.interceptor;
-
-import io.openmessaging.Message;
-
-/**
- * A {@code ProducerInterceptor} is used to intercept send operations of producer.
- * <p>
- * The interceptor is able to view or modify the message being transmitted and collect
- * the send record.
- *
- * @version OMS 1.0.0
- * @since OMS 1.0.0
- */
-public interface ProducerInterceptor {
```

Comment:
IMO, I don't think we should remove this interface.  and instead with a MessageListener, because MessageInterceptor.onMessage(Message) can be invoked before a message was sent or after a message was sent.

Reply:
These two new added methods in the interface Producer solved the problem: 
```
    /**
     * Adds a {@code MessageInterceptor} to intercept send operations of producer.
     *
     * The interceptor will be invoked before the message is actually sent to the network.
     * <p>
     * This allows for modification of the message if necessary.
     * @param interceptor a message interceptor.
     */
    void addPreSendInterceptor(MessageInterceptor interceptor);

    /**
     * Adds a {@code MessageInterceptor} to intercept send operations of producer.
     *
     * The interceptor will be invoked immediately after the successful send invocation.
     * @param interceptor a message interceptor.
     */
    void addPostSendInterceptor(MessageInterceptor interceptor);
```